src/content/guide/tools-and-features/system-firmware.md

```diff
@@ -0,0 +1,170 @@
+
+Application firmware that is written in the [Web](https://build.particle.io) or [Desktop](https://www.particle.io/products/development-tools/particle-desktop-ide) IDEs are _compiled against_ a specific version of system firmware before being sent to a device to run. That is, the system firmware acts as a translator - taking the human-readable code you write using the firmware API and translating into machine-executable code that the device can run.
+
+This creates a dependency that must be carefully managed. For example, imagine a new firmware primitive was introduced in system firmware version `1.0.0`, `Particle.travelInTime()`. As an aspiring time traveler, you quickly add the new feature to your firmware logic and send the code off to be compiled and flashed to your Electron.
+
+However, the Electron on your desk is running system firmware `0.9.0`, a version that predates the time travel functionality. Instructing the device to use the new firmware method in application firmware before it understands how to do so will of course not work. You can see how application firmware _depends on_ a compatible version of system firmware.
+
+So what happens in these cases?
+
+### Safe Mode
+When booting up, the Particle device will check dependencies between the application and the system firmware. In the case of an incompatibility between the two, the device will automatically enter into [_safe mode_](/guide/getting-started/modes/#safe-mode) (breathing magenta). 
+
+Safe mode allows the device to connect to the Particle cloud, but does not run application firmware. There are many uses for safe mode, but is particularly relevant when a device receives application firmware compiled against a newer version of system firmware than it currently is running. In this case, safe mode prevents the device from running the incompatible application firmware that will cause it to hard fault.
+
+### Safe Mode Healer
+_Safe mode healer_ takes things one step further, automatically resolving system firmware incompatibilities when a device enters safe mode. Because a device in safe mode still has network connectivity, it is able to send an event to the Particle Cloud notifying it of the firmware mismatch.
+
+From this event, the Particle Cloud determines the cause of the incompatibility and delivers to the device new system firmware over-the-air. When this happens, you'll see the device rapidly blinking magenta as it receives packets containing the new system firmware. When complete, the device will have what it needs to successfully run the application that was previously flashed.
+
+The combination of safe mode and safe mode healer provides you confidence when flashing new application firmwares to devices. Regardless of the version of system firmware the app was compiled against, the Particle Cloud will ensure the device has what it needs to run it without issue.
+
+A couple of important notes:
```

Comment:
These notes are too late in the guide IMOH. I essentially rewrote these as part of the text above.